### DISCUSSION TITLE
Access to <extensions> elements in MD block

### DISCUSSION THREAD

**Discussion by radix-trie** [danielhrisca/asammdf#993] [Q&A] [answered]
Hello,

I have encountered issue, when trying to access `<extensions>` content of MD block. In readme, there is note about `xml schema for MDBLOCK: most metadata stored in the comment blocks will not be available`, but in older versions of asammdf (5.x.x) I could access the content of header MD block as plain text and parse the extensions as I needed, in current version (7.4.1) everything outside of common properties is omitted, once the instance is created.

Is this intentional, missing implementation, or defect? Is it possible, to access full content of MD block (i.e.: as string) when I load `.mf4` file in python script in some "clean way"?

Kind regards,
Rt.

**Comment by danielhrisca** [comment]
Please give an example

**Comment by radix-trie** [comment]
Hello,

Example of the text extracted via `get_text_v4` which is then used in HeaderBlock class here (based on my debugging): https://github.com/danielhrisca/asammdf/blob/edaac991f183603383c274ec214bbc32e6793e32/src/asammdf/blocks/v4_blocks.py#L5557

I will try to provide minimal mf4 file later. The example of text is:

```
<HDcomment>
    <TX>Manual trigger</TX>
    <time_source>External Time Source</time_source>
    <common_properties>
        <e name="author">Author</e>
        <e name="department">Department</e>
        <e name="project">Project</e>
        <e name="subject">Subject</e>
    </common_properties>
    <extensions>
        <extension ci="our-tool" xmlns:v="http://company/project/tool-space">
            <v:attributeA>StringValue</v:attributeA>
            <v:attributeB>25000</v:attributeB>
            <v:attributeC>
                <v:value>27.523</v:value>
            </v:attributeC>
            <v:attributeD>
                <v:value>42.000</v:value>
            </v:attributeD>
        </extension>
    </extensions>
</HDcomment>
```

**Reply by danielhrisca** [reply, marked as answer]
Please try the development branch code

**Reply by radix-trie** [reply]
Yes! with dev version I can see the full content of header, namespaces seems to be shuffled around a bit, but that is just a result of the processing via `etree`. Thank you very much for quick reaction!
